src/type_parsers/datetime_parser.jl

```diff
@@ -217,6 +266,13 @@ end
 # This is needed until https://github.com/JuliaTime/TimeZones.jl/issues/271 is fixed
 const _Z = SubString("Z", 1:1)
 @inline function _tryparse_timezone(buf, pos, b, len, code)
+    # At this point we don't even know if there is a timezone to parse, we might be at the end of
+    # the field. So in case we get an invalid TZ here, we just return the _original_ code
+    # and `nothing` for the timezone, as if we never attempted to parse it.
+    # If this _was_ a true invalid timezone, the other layers in Parsers.jl will mark the value
+    # as invalid because were at the very end of the field and if leave any non-whitespace characters
+    # between the end of the value and the delimiter, it will be marked as invalid by other layers
```

Comment:
typos here? not sure i understood tbh 😝 
```suggestion
    # as invalid because we're at the very end of the field and if we leave any non-whitespace characters
    # between the end of the value and the delimiter, it will be marked as invalid by other layers
```